### DISCUSSION TITLE
Creating item from specific user in custom endpoint (no accountability)

### DISCUSSION THREAD

**Discussion by iksent** [directus/directus#18656] [Archive]
### Subject Area

Maybe API Reference...

#### I have the following scheme in my project:
- directus_users
  - experts
    - skills

#### Brief outline of my problem:
- Custom endpoint of **Registration**
  - user created
    - expert created
      - skills created - **BUT**: no accountability to correctly set **user_created** field

#### Detailed scheme and problem
There are users and there are experts. For experts, a separate table has been created, which is associated with M2O user (1 expert = 1 user). With the help of a custom endpoint, a user is created during registration and an Expert is attached to it. During registration, other data (skills) are also created, which are associated with the expert.
With the help of the rights and the role of an expert, it is regulated that an expert can only edit his own skills.
The peculiarity of the scheme is that skills are created immediately during registration and creation of an Expert, at this moment the user is not yet authorized and there is no specific accountability in the endpoint.

**Problem**: when creating skills, I need to set the user_created fields (so that only this expert can change them later and only them), but I do not have specific **accountability** in the endpoint, as user not authorized during registration.

**Question**: What to do and how to solve this issue? And how can we improve the documentation for this scenario?

**Comment by iksent** [comment]
1. I tried to pass `user_created` field with newly created user's ID to `skills` object. No success.

- I found such solution: https://github.com/directus/directus/discussions/3015#discussioncomment-126094
- BUT: I just have no such field to change in `user_created` config... I don't know why so.

2. Also I tried to fetch permissions for Expert role and use them at `accountability` object. Also Forbidden error.

```
     const permissionService = new PermissionsService({
        schema: req.schema,
      })
      const permissions = await permissionService.readByQuery({
        fields: ['*.*.*'],
        filter: {
          role: EXPERT_ROLE_ID,
        },
      })
      const newAccountability = {
        ...req.accountability,
        user: createdUserID,
        role: EXPERT_ROLE_ID,
        permissions,
      }
```

3. Maybe I should try to login as newly created user? But where to get `accountability` after this?

**Reply by connorwinston** [reply]
@iksent Can you share what the field config is for `user_created`? Your (1.) should work.

**Reply by iksent** [reply]
> @iksent Can you share what the field config is for `user_created`? Your (1.) should work.

Are you talking about config in snapshot?

About 1.: But what if I'd like to save auto-setting of user_created field on item creation?

**Reply by connorwinston** [reply]
> Are you talking about config in snapshot?

How do you have the field set up in the data model? AKA what the field's type is set to, is it set to auto set on creation, etc.?

If it is set to save user_created on item creation, then it will use the [user field on the accountability](https://github.com/directus/directus/blob/85c664f9853dc8e3a169bf6a9931a5cd30bbd923/api/src/services/payload.ts#L106). If you are using ItemService you will need to create the new service with the new accountability. If you have custom validation logic and permission checking in your endpoint, you should, in theory, be able to use `{ user: USER_ID, admin: true}` as the accountability object, which will not need permissions in there.

**Reply by iksent** [reply]
Oh my! `admin: true` really helped to fix all my problems.
Thanks a lot, @ConnorSimply!
